test/jest/dep-graph-builders/yarn-lock-v2.test.ts

```diff
@@ -1,58 +1,444 @@
 
-describe('dep-graph-builder yarn-lock-v1', () => {
-  describe('happy path tests', () => {
-    describe('Expected Result tests', () => {
-      describe.each([
-        'git-remote-url',
-        'goof',
-        'resolutions-simple',
-        'resolutions-scoped',
-      ])('[simple tests] project: %s ', (fixtureName) => {
-        test('matches expected', async () => {
-          const pkgJsonContent = readFileSync(
-            join(
-              __dirname,
-              `./fixtures/yarn-lock-v2/${fixtureName}/package.json`,
-            ),
-            'utf8',
-          );
-          const yarnLockContent = readFileSync(
-            join(__dirname, `./fixtures/yarn-lock-v2/${fixtureName}/yarn.lock`),
-            'utf8',
-          );
-          const opts = {
-            includeDevDeps: false,
-            includeOptionalDeps: true,
-            pruneCycles: false,
-            strictOutOfSync: false,
-          };
-
-          const dg = await parseYarnLockV2Project(
-            pkgJsonContent,
-            yarnLockContent,
-            opts,
-          );
-
-          const expectedDepGraphJson = JSON.parse(
-            readFileSync(
-              join(
-                __dirname,
-                `./fixtures/yarn-lock-v2/${fixtureName}/expected.json`,
-              ),
-              'utf8',
-            ),
-          );
-
-          expect(dg).toBeTruthy();
-
-          const expectedDepGraph = createFromJSON(expectedDepGraphJson);
-          expect(dg.equals(expectedDepGraph)).toBeTruthy();
-        });
-      });
+describe('yarn.lock v2 "real" projects', () => {
+  describe.each([
+    'git-remote-url',
+    'goof',
+    'resolutions-simple',
+    'resolutions-scoped',
+  ])('[simple tests] project: %s ', (fixtureName) => {
+    test('matches expected - no pruning', async () => {
+      const pkgJsonContent = readFileSync(
+        join(
+          __dirname,
+          `./fixtures/yarn-lock-v2/real/${fixtureName}/package.json`,
+        ),
+        'utf8',
+      );
+      const yarnLockContent = readFileSync(
+        join(
+          __dirname,
+          `./fixtures/yarn-lock-v2/real/${fixtureName}/yarn.lock`,
+        ),
+        'utf8',
+      );
+      const opts: YarnLockV2ProjectParseOptions = {
+        includeDevDeps: false,
+        includeOptionalDeps: true,
+        strictOutOfSync: false,
+        pruneWithinTopLevelDeps: false,
+      };
+
+      const dg = await parseYarnLockV2Project(
+        pkgJsonContent,
+        yarnLockContent,
+        opts,
+      );
+
+      const expectedDepGraphJson = JSON.parse(
+        readFileSync(
+          join(
+            __dirname,
+            `./fixtures/yarn-lock-v2/real/${fixtureName}/expected.json`,
+          ),
+          'utf8',
+        ),
+      );
+
+      expect(dg).toBeTruthy();
+
+      const expectedDepGraph = createFromJSON(expectedDepGraphJson);
+      expect(dg.equals(expectedDepGraph)).toBeTruthy();
```

Comment:
test fixtures didn't change! does that mean non of these real test fixtures had pruning opportunities, and/or does this equals not care about prune labels

Reply:
Must be the former

Reply:
i can't believe goof cannot be pruned

Reply:
The old tests have pruning off in the options, it's a flag you pass in this eco

Reply:
https://github.com/snyk/nodejs-lockfile-parser/pull/191/files#diff-c09b371ecae0c6d79c228da21ef8f9ae6932a8ccd45472b23a1b60b8799eaae7R29-R34

Reply:
are we ever going to not prune? whats the value in those tests

Reply:
I believe we should test both, this library may be used in applications that don not wish to prune